lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -440,92 +457,239 @@ void LowerClassesPass::lowerClassExtern(ClassExternOp classExternOp,
   }
 }
 
-// Update Object instantiations in a FIRRTL Module or OM Class.
-LogicalResult LowerClassesPass::updateInstances(Operation *op) {
-  OpBuilder builder(op);
-  // For each Object instance.
-  for (auto firrtlObject : llvm::make_early_inc_range(
-           op->getRegion(0).getOps<firrtl::ObjectOp>())) {
-
-    // build a table mapping the indices of input ports to their position in the
-    // om class's parameter list.
-    auto firrtlClassType = firrtlObject.getType();
-    auto numElements = firrtlClassType.getNumElements();
-    llvm::SmallVector<unsigned> argIndexTable;
-    argIndexTable.resize(numElements);
-
-    unsigned nextArgIndex = 0;
-    for (unsigned i = 0; i < numElements; ++i) {
-      auto direction = firrtlClassType.getElement(i).direction;
-      if (direction == Direction::In)
-        argIndexTable[i] = nextArgIndex++;
-    }
+// Helper to update an Object instantiation. FIRRTL Object instances are
+// converted to OM Object instances.
+static LogicalResult
+updateObjectInstance(firrtl::ObjectOp firrtlObject, OpBuilder &builder,
```

Comment:
This should be the same implementation as what was previously in the loop body of `updateInstances`. While resolving merge conflicts, I just copy-pasted the implementation currently on main. The only change I made was to delay erasing ops, so the main loop in updateInstances wouldn't visit an up that had already been deleted.